MetaMask/metamask-extension
# Fix gas api overcalling

**Fixes: #12002 & #12067**

**Explanation**:  I introduced a bug in this [PR](https://github.com/MetaMask/metamask-extension/pull/11936) in which we (way) over call to our gasprice estimate API. This fixes this issue by moving the gas API calls into more appropriate places.

(edit) **More context**:
The PR that introduced the bug was meant to improve our speedup/cancel transaction success rate by introducing gasEstimates as part of the calculation for what speedup/cancel gas fee's should be set to. This was poorly implemented (by me) and introduced a call to the gas Estimates API (via useGasFeeEstimates) in a hook (useIncrementedGasFees) that was called directly on each TransactionListItem in the activity log.

This PR relocates and compartmentalizes the calls to the API ( via useGasFeeEstimates) in such a way that no functionality is lost but only required API calls are made.

**Manual testing steps:**  
  - Open the background console and navigate tot he network tab
  - Send a simple eth transaction on Rinkeby (or Mainnet)
  - Make sure there are not a ton of queries to `SuggestedGasFee` (request URL: https://gas-api.metaswap.codefi.network/networks/4/suggestedGasFees)
  - There should be 1 or 2 every ~10 seconds and no more than that.
